### PR TITLE
[Doc] Update contribution guide

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

This PR updates the contribution guide with revisions based on the current workflow.

## Related Issue
Closes PWA-122

## Acceptance 

@tjwiebell 
@awilcoxa 
@Jordaneisenburger (optional)

### Verification Stakeholders

@tjwiebell 
@awilcoxa 
@Jordaneisenburger (optional)

### Verification Steps

1. Open `.github/CONTRIBUTING.md`
1. Verify content is accurate

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,134 +1,93 @@
...
-In general this meeting is brief but can vary with the complexity of an issue.
-For larger issues, we may schedule follow-up meetings.
+#### Grooming new issues
 
-During this meeting, we provide you with any additional materials or resources you need to work on the issue.
+Every week, the Community Advisory Board meets to look at the **Backlog** column of the Community Backlog project.
+The board grooms the backlog issues and moves issues that are well-defined and has value to the **Prioritized** column.
 
-#### Step 3: Provide an estimate
+If an issue does not provide enough details or provides no value, it is closed.
```
Comment (Jordaneisenburger) [inline .github/CONTRIBUTING.md]: @jcalcaben If we feel that an issue does not provide enough detail I think we shouldn't close it but rather leave a comment and give them the opportunity to respond if there is no reaction before the next grooming session we should indeed close it.

Comment (jcalcaben) [reply .github/CONTRIBUTING.md]: Updated statement

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,134 +1,94 @@
...
-#### Step 1: Add an issue comment
+### Identify and create issues
 
-Add a comment on an issue expressing your interest in taking ownership of it.
-Make sure your GitHub profile includes an email address so we can contact you privately.
+If you encounter an issue while using PWA Studio or you have a suggestion for a new feature, let us know by [creating an issue][].
+Provide as much detail as you can to help us reproduce or analyze the issue before prioritizing it.
 
-#### Step 2: Meet with a maintainer
+When you create an issue, add the **help wanted** label to bring it to the attention of the Community Advisory Board.
```
Comment (tjwiebell) [inline .github/CONTRIBUTING.md]: Community members will not have the ability to add this label, so I would say to ping us in Slack, or just say we will go through newly created issues and add the label to issues we deem suitable for the community.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,134 +1,94 @@
...
+To get your contribution accepted, you must sign [Adobe's Contributor License Agreement (CLA)](https://opensource.adobe.com/cla.html).
+You only need to sign the CLA once.
 
-After we accept and merge your contribution, you become an official contributor!
-Official contributors are invited to our backlog grooming sessions and have direct influence over the product roadmap.
+#### PR review process
 
-We hope this guide paints a clear picture of your duties and expectations in the contribution process. Thank you in advance for helping with our research projects!
+After you submit your PR, the Community Advisory Board will assign one or more reviewers to look over your solution and ensure quality and adherence to standards.
+As the Directly Responsible Individual for the issue, you are expected to address all feedback before the issue is sent to our QA and merged.
```
Comment (tjwiebell) [inline .github/CONTRIBUTING.md]: I would add something here that makes this sound less overwhelming to new developers.

> If at any point in the review cycle you have commitments preventing you from addressing feedback, or if there's anything suggested you're not understanding, please let us know; we're happy to pair program or complete those changes for you to push PRs across the finish line.

**Pull request merged**